performance/multi-node/random_access.c

```diff
@@ -0,0 +1,1081 @@
+	}
+
+	/* Open domain */
+	ret = fi_domain(fabric->fab, fabric->info, &fabric->dom, NULL);
+	if (unlikely(ret)) {
+		ct_print_fi_error("fi_domain", ret);
+		goto err_domain_open;
+	}
+
+	return 0;
+
+	err_domain_open: fi_close(&fabric->fab->fid);
+	err_fabric_open: return ret;
+}
+
+static int init_endpoint(fabric_t *fabric, int remote_rank)
+{
+	struct peer *p = &fabric->peers[remote_rank];
+	int ret;
+	int i;
+
+	/* Open endpoint */
+	ret = fi_endpoint(fabric->dom, fabric->info, &p->ep, NULL);
+	if (ret) {
+		ct_print_fi_error("fi_endpoint", ret);
+		goto err_fi_endpoint;
+	}
+
+	/* Allocate endpoint resources */
+	ret = alloc_ep_res(fabric, remote_rank);
+	if (ret)
+		goto err_alloc_ep_res;
+
+	/* Bind EQs and AVs with endpoint */
+	ret = bind_ep_res(fabric, remote_rank);
+	if (ret)
+		goto err_bind_ep_res;
+
+	return 0;
+
+	err_bind_ep_res: free_ep_res(fabric, remote_rank);
+	err_alloc_ep_res: fi_close(&p->ep->fid);
+	err_fi_endpoint: return ret;
+}
+
+static int init_av(fabric_t *fabric)
+{
+	void *addr;
+	size_t addrlen = 0;
+	int ret;
+	int i;
+	struct peer *peers = fabric->peers;
+
+	/* use first ep since they will all have the same source addr */
+	for (i = 0; i < num_ranks; i++)
+		if (fabric->peers[i].ep)
+			break;
+
+	fi_getname(&peers[i].ep->fid, NULL, &addrlen);
+
+	addr = malloc(addrlen);
+	assert(addr);
+
+	ret = fi_getname(&peers[i].ep->fid, addr, &addrlen);
+	if (ret != 0) {
+		ct_print_fi_error("fi_getname", ret);
+		free(addr);
+		return ret;
+	}
+
+	fabric->addrs = calloc(num_ranks, addrlen);
+	assert(fabric->addrs);
+
+	ctpm_Allgather(addr, addrlen, fabric->addrs);
+
+	fabric->fi_addrs = calloc(num_ranks, sizeof(fi_addr_t));
+	assert(fabric->fi_addrs);
+
+	/* Insert address to the AV and get the fabric address back */
+	ret = fi_av_insert(fabric->av, fabric->addrs, num_ranks, fabric->fi_addrs,
+			0, NULL);
+	if (ret != num_ranks) {
+		ct_print_fi_error("fi_av_insert", ret);
+		return ret;
+	}
+
+	free(addr);
+
+	return 0;
+}
+
+static int64_t generate_initial_xor_val(uint64_t seed)
+{
+	/* int64_t i, j; */
+	int i, j;
+	uint64_t m2[64];
+	uint64_t temp, ret;
+
+	/* This section of code is legacy code from when seed used to be a
+	 * signed int
+	 while (n < 0)
+	 n += PERIOD;
+	 */
+	while (seed > PERIOD)
+		seed -= PERIOD;
+	if (seed == 0)
+		return 0x1;
+
+	temp = 0x1;
+	for (i = 0; i < 64; i++) {
+		m2[i] = temp;
+		temp = (temp << 1) ^ ((int64_t)temp < 0 ? POLY : 0);
+		temp = (temp << 1) ^ ((int64_t)temp < 0 ? POLY : 0);
+	}
+
+	for (i = 62; i >= 0; i--)
+		if ((seed >> i) & 1)
+			break;
+
+	ret = 0x2;
+
+	while (i > 0) {
+		temp = 0;
+		for (j = 0; j < 64; j++)
+			if ((ret >> j) & 1)
+				temp ^= m2[j];
+		ret = temp;
+		i -= 1;
+		if ((seed >> i) & 1)
+			ret = (ret << 1) ^ ((int64_t)ret < 0 ? POLY : 0);
+	}
+
+	return ret;
+}
+
+static void update_table(uint64_t *table, uint64_t table_size,
+		uint64_t min_local_table_size, uint64_t top, int remainder,
+		int64_t iterations, int use_lock)
```

Comment:
Nope. Another cleanup item. That was present before I was doing the operations with BXOR.